src/rust/src/http_reader.rs

```diff
@@ -27,8 +27,33 @@ pub struct HttpFgbReader {
 
 impl HttpFgbReader {
     pub async fn open(url: &str) -> Result<HttpFgbReader> {
+        trace!("starting: opening http reader, reading header");
         let mut client = BufferedHttpClient::new(&url);
-        let min_req_size = 512;
+
+        // Because we use a buffered HTTP reader, anything extra we fetch here can
+        // be utilized to skip subsequent fetches.
+        // Immediately following the header is the optional spatial index, we deliberately fetch
+        // a small part of that to skip subsequent requests
+        let prefetch_index_bytes: usize = {
+            // The actual branching factor will be in the header, but since we don't have the header
+            // yet we guess. The consequence of getting this wrong isn't catastrophic, it just means
+            // we may be fetching slightly more than we need or that we make an extra request later.
+            let assumed_branching_factor: usize = 16;
+
+            // NOTE: each layer is exponentially larger
+            let prefetched_layers: u32 = 3;
+
+            (0..prefetched_layers).map(|i| {
+                assumed_branching_factor.pow(i) * std::mem::size_of::<NodeItem>()
```

Comment:
Should that be the same as level node counts from https://github.com/bjornharrtell/flatgeobuf/blob/master/src/rust/src/packed_r_tree.rs#L300 ?

Reply:
Right, basically - but this is an intentional over-estimate since we don't yet have the real `item_count`.

If the actual item_count happened to be a power of 16, then I believe they should match exactly.